### PR TITLE
fix: check against valid operations before checking parameters exist.

### LINKED ISSUE #1187: Path Params being applied to more than http operations
**Describe the bug**
the `path-params` rule is being applied to more than just http operations defined under a path.

**To Reproduce**

```
openapi: 3.0.1
paths:
  /hello/{id}:
    x-dont-path-param-me: hello
    get:
      tags:
        - metadata
      operationId: list-data-sets
      summary: List available data sets
      responses:
        '200':
          description: Returns a list of data sets
      parameters:
        - name: id
          in: path
          required: true
          schema:
            type: string
```

output:
```
  1:1  warning  info-contact           Info object should contain `contact` object.
  1:1  warning  info-description       OpenAPI object info `description` must be present and non-empty string.
  1:1  warning  oas3-api-servers       OpenAPI `servers` must be present and non-empty array.
  1:1    error  oas3-schema            Object should have required property `info`.
  1:1  warning  openapi-tags           OpenAPI object should have non-empty `tags` array.
 4:27    error  path-params            The operation does not define the parameter `{id}` expected by path `/hello/{id}`.
  5:9  warning  operation-description  Operation `description` must be present and non-empty string.
 7:11  warning  operation-tag-defined  Operation tags should be defined in global tags.
```

you can see here that the path params rule is being applied to `x-dont-path-param-me`

**Expected behavior**
I expect it to only be applied to the standard http operations (get, post, put, delete, options, etc)

**Screenshots**

**Environment (remove any that are not applicable):**
 - Library version: 5.4.0
 - OS: MacOS 10.15.3

### PR DESCRIPTION
**Pull request by bzmw**
Fixes #1187.

**Checklist**

- [X] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

> If indicated yes above, please describe the breaking change(s).
>
> **Remove this quote before creating the PR.**

**Screenshots**

If applicable, add screenshots or gifs to help demonstrate the changes. If not applicable, remove this screenshots section before creating the PR.

**Additional context**

Add any other context about the pull request here. Remove this section if there is no additional context.

### CODE REVIEW

**Comment by nulltoken** [comment]
@bzmw That looks very good to me. Awesome! 

@P0lip Thoughts?

File: src/rulesets/oas/functions/oasPathParam.ts [hunk 1]
```diff
@@ -103,6 +103,7 @@ export const oasPathParam: IFunction = (targetVal, _options, paths, vals) => {
 
   // keep track of normalized paths for verifying paths are unique
   const uniquePaths: object = {};
+  const validOperationKeys = ['get', 'head', 'post', 'put', 'patch', 'delete', 'options', 'trace'];
```
Comment (nulltoken) [inline src/rulesets/oas/functions/oasPathParam.ts:106]: @bzmw Could you please also update the `path-params` section of `openapi-rules.md` doco to quickly explain that very non standard HTTP verb will be silently ignored?

Comment (bzmw) [reply src/rulesets/oas/functions/oasPathParam.ts:106]: Done!

**Comment by P0lip** [comment]
Thanks again!

**Pull request merged**